pkg/controllers/mcs/mcs_serviceexport_controller.go

```diff
@@ -0,0 +1,200 @@
+package mcs
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	mcsv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	networkingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/events"
+	"github.com/karmada-io/karmada/pkg/sharedcli/ratelimiterflag"
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+const (
+	// MultiClusterServiceControllerName MultiClusterServiceController name.
+	MultiClusterServiceControllerName = "multi-cluster-service-controller"
+
+	// MultiClusterServiceControllerFinalizer is added to MultiClusterService to ensure related Works are deleted
+	// before MultiClusterService itself is deleted.
+	MultiClusterServiceControllerFinalizer = "karmada.io/multi-cluster-service-controller"
```

Comment:
How about modifying this finalizer like this?
```go
// karmadaMCSFinalizer is added to MultiClusterService to ensure downstream resource,
// such as ServiceExport and ServiceImport are deleted before itself is deleted.
const karmadaMCSFinalizer = "karmada.io/karmada-mcs-controller"
```

In `multicluster-cloud-provider` repo, we also need to add similar finalizers, for example.

```go
const providerMCSFinalizer = "karmada.io/multicluster-cloud-provider-mcs-controller"
```

How do you think?